langchain/document_loaders/base.py

```diff
@@ -1,14 +1,17 @@
 """Base loader class."""
+from __future__ import annotations
 
+import abc
 from abc import ABC, abstractmethod
-from typing import List, Optional
+from typing import List, Optional, Generator
 
-from langchain.docstore.document import Document
+from langchain.document_loaders.blob_loaders import Blob
```

Comment:
perhaps BlobParser should live in blob_loaders?

Reply:
Yeah we can definitely move the `BlobParser`, we'll need to decide what's the most logical place

One thing I wasn't so sure about is that the document loaders right now are basically content fetchers + parsers and the majority of their work should be focused on parsing, so it felt like a parser abstraction could be accommodated at this level of the hierarchy. I'll start diffing smaller more careful changes now, and we can iron out all the namespaces to make sure everything is in the most logical place. 

`base` aka `typedefs` ideally don't import much at all (except for other typedefs), so there's that...

IMO it's generally OK to import more nested code, since the nested code is owned by the importing code. The dangerous imports are when reaching into parent paths or sibling paths since that means that we're reaching into nested code that doesn't belong to the given module/package